interactive_engine/executor/ir/core/src/catalogue/mod.rs

```diff
@@ -42,3 +43,12 @@ pub(crate) fn query_params(
         extra: HashMap::new(),
     }
 }
+
+pub trait PatternOrderTrait<D> {
+    fn compare(&self, left: &D, right: &D) -> Ordering;
+}
+
+pub trait PatternWeightTrait<W: PartialOrd> {
+    fn estimate_vertex_weight(&self, vid: PatternId) -> W;
```

Comment:
`get_vertex_weight()` and `get_adjacencies_weight()` are better names.

Reply:
Done